burnman/solutionmodel.py

```diff
@@ -326,3 +323,81 @@ def __init__( self, endmembers, enthalpy_interaction, volume_interaction = None,
         alphas = np.ones( len(endmembers) )
         AsymmetricRegularSolution.__init__(self, endmembers, alphas, enthalpy_interaction, volume_interaction, entropy_interaction )
 
+class SubregularSolution (IdealSolution):
+    """
+    Solution model implementing the subregular solution model formulation (Helffrich and Wood, 1989)
+    """
+
+    def __init__( self, endmembers, enthalpy_interaction, volume_interaction = None, entropy_interaction = None ): 
+
+        self.n_endmembers = len(endmembers)
+
+        # Create 2D arrays of interaction parameters
+        self.Wh=np.zeros(shape=(self.n_endmembers,self.n_endmembers))
+        self.Ws=np.zeros(shape=(self.n_endmembers,self.n_endmembers))
+        self.Wv=np.zeros(shape=(self.n_endmembers,self.n_endmembers))
+
+        #setup excess enthalpy interaction matrix
+        for i in range(self.n_endmembers):
+            for j in range(i+1, self.n_endmembers):
+                self.Wh[i][j]=enthalpy_interaction[i][j-i-1][0]
+                self.Wh[j][i]=enthalpy_interaction[i][j-i-1][1]
+
+        if entropy_interaction is not None:
+            for i in range(self.n_endmembers):
+                for j in range(i+1, self.n_endmembers):
+                    self.Ws[i][j]=entropy_interaction[i][j-i-1][0]
+                    self.Ws[j][i]=entropy_interaction[i][j-i-1][1]
+
+        if volume_interaction is not None:
+            for i in range(self.n_endmembers):
+                for j in range(i+1, self.n_endmembers):
+                    self.Wv[i][j]=volume_interaction[i][j-i-1][0]
+                    self.Wv[j][i]=volume_interaction[i][j-i-1][1]
+
+        #initialize ideal solution model
+        IdealSolution.__init__(self, endmembers)
+
+    def _non_ideal_function(self, W, molar_fractions):
```

Comment:
Well, this is how it's written in the original paper. I'm not sure it would be more readable in terms of matrix operations, but I can have a look if you think it's worth it?